earth_enterprise/src/fusion/autoingest/Asset.h

```diff
@@ -78,15 +78,13 @@ class AssetImpl : public AssetStorage, public StorageManaged {
   const SharedString & GetRef(void) const { return name; }
 
   // determine amount of memory used by an AssetImpl
-  uint64 GetSize() {
-    return(GetObjectSize(name)
-    + GetObjectSize(type)
-    + GetObjectSize(subtype)
-    + GetObjectSize(inputs)
-    + meta.GetSize()
-    + GetObjectSize(versions)
-    + GetObjectSize(timestamp)
-    + GetObjectSize(filesize));
+  virtual uint64 GetHeapUsage() const {
+    return ::GetHeapUsage(name)
```

Comment:
Any reason some of the members are missing here (i.e., `type`, `filesize`)? Even if they don't use the heap you can still call GetHeapUsage and it will return 0.

Reply:
Looking at some of the other classes it looks like you're just not calling GetHeapSize for primitives. I can live with that.